README.md

```diff
@@ -3,3 +3,95 @@
 
 This package provides a CUDA implementation for many of the modules in the base nn package: [nn](https://github.com/torch/nn/blob/master/README.md)
  * [Modules](doc/cunnmodules.md#nn.cunnmodules.dok): There are also additional GPU-related modules not found in the nn package.
+
+## To use
+
+Simply convert your network model to CUDA by calling `:cuda()`:
+```lua
+local model = nn.Sequential()
+model:add(nn.Linear(2,2))
+model:add(nn.LogSoftMax())
+
+model:cuda()  -- convert model to CUDA
+```
+
+... and similarly for your tensors:
+```lua
+local input = torch.Tensor(32,2):uniform()
+input = input:cuda()
+local output = model:forward(input)
+```
+... or create them directly as `CudaTensor`s:
+```lua
+local input = torch.CudaTensor(32,2):uniform()
+local output = model:forward(input)
+```
+
+## To run unit-tests
+
+```lua
+luajit -l cunn -e 'cunn.test()'
+```
+
+## GPU Training Concepts
+
+__Performance__
+
+* data should be transferred between main memory and gpu in batches, otherwise the transfer time will be dominated
+by latency associated with speed of light, and execution overheads, rather than by bandwidth
+* therefore, train and predict using mini-batches
+* allocating GPU memory causes a sync-point, which will noticeably affect performance
+  * therefore try to allocate any `CudaTensor`s once, at the start of the program,
+  and then simply copy data backwards and forwards
+  between main memory and existing `CudaTensor`s
+* similarly, try to avoid any operations that implicitly allocate new tensors.  For example, if you write:
+```lua
+require 'cutorch'
+
+local a = torch.CudaTensor(1000):uniform()
+for it=1,1000 do
+  local b = a:add(1)
```

Comment:
This is a bad example IMHO, as in here `a:add(1)` returns `a`, meaning that there is no memory allocation (`a` and `b` are the same lua object).
Maybe it would be better something like `b = torch.add(a,1)` or `b = a + 1` ?